sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -176,6 +176,8 @@ class RegexParser(pattern: String) {
                 case Some(end) =>
                   skip()
                   characterClass.appendRange(start, end)
+                case _ =>
+                  // TODO https://github.com/NVIDIA/spark-rapids/issues/4075
```

Comment:
The code used to throw a `MatchError` but now will silently suppress it.  If we're waiting for #4075 to fix this the right way then this should throw `IllegalStateException`, `MatchError`, `AssertionException` or some kind of stand-in exception so we don't silently eat this case with a PR that shouldn't be changing the code semantics.  cc: @andygrove for visibility in case this should simply throw `RegexUnsupportedException` (not sure what the message arg should be in that case).

Reply:
The code is currently matching `Some(']')` or `Some(_)` and is missing `None` which could happen with a malformed regular expression. I would suggest adding this:

```scala
case None =>
  throw new RegexUnsupportedException("unexpected EOF while parsing character range", Some(pos))
```

Reply:
I am aware of the semantics, and incorporated the change. The thought was that it won't take long for the follow-on

Reply:
Understood.  I've seen too many followup PRs that don't get addressed as timely a we think they will, and a PR with this title shouldn't be the result of a `git bisect` to figure out what in the world changed to cause some bizarre behavior. :smile: